File: src/models/games/ff_game/ff_ball.rs
use serde::{Deserialize, Serialize};

use crate::models::geometry::square::Square;

#[derive(Default, Clone, Eq, PartialEq, Debug, Serialize, Deserialize)]
pub struct FfBall {
    square: Square,
    in_game: bool,
    under_control: bool,
    on_ground: bool
}

impl FfBall {
	pub fn new() -> FfBall {
        Default::default()
    }    
}

File: java/src/main/java/com/foul/gameserver/games/FfGame/FfBall.java
package com.foul.gameserver.games.FfGame;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import com.foul.gameserver.domain.geometry.Square;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.ToString;

@Getter
@EqualsAndHashCode
@RequiredArgsConstructor
@ToString
public final class FfBall {

    Logger logger = LoggerFactory.getLogger(FfBall.class);

    Square square;
    boolean inGame;
    boolean underControl;
    boolean onGround;

}
